@xen-orchestra/web-core/lib/components/ui/button/UiButton.vue

```diff
@@ -0,0 +1,472 @@
+<!-- v4 -->
+<template>
+  <button :class="classNames" :disabled="busy || isDisabled" class="ui-button" type="button">
+    <VtsIcon :busy :icon="leftIcon" accent="current" class="icon" fixed-width />
+    <slot />
+    <VtsIcon :icon="rightIcon" accent="current" class="icon" fixed-width />
+  </button>
+</template>
+
+<script lang="ts" setup>
+import VtsIcon from '@core/components/icon/VtsIcon.vue'
+import { useContext } from '@core/composables/context.composable'
+import { DisabledContext } from '@core/context'
+import { toVariants } from '@core/utils/to-variants.util'
+import type { IconDefinition } from '@fortawesome/fontawesome-common-types'
+import { computed } from 'vue'
+
+type ButtonVariant = 'primary' | 'secondary' | 'tertiary'
+type ButtonAccent = 'brand' | 'success' | 'warning' | 'danger'
+type ButtonSize = 'small' | 'medium' | 'large'
+
+const props = withDefaults(
+  defineProps<{
+    variant: ButtonVariant
+    accent: ButtonAccent
+    size: ButtonSize
+    busy?: boolean
+    disabled?: boolean
+    leftIcon?: IconDefinition
+    rightIcon?: IconDefinition
+  }>(),
+  {
+    disabled: undefined,
+  }
+)
+
+defineSlots<{
+  default(): any
+}>()
+
+const isDisabled = useContext(DisabledContext, () => props.disabled)
+
+const fontClasses = {
+  small: 'typo p3-medium',
+  medium: 'typo h6-medium',
+  large: 'typo h3-semi-bold',
+}
+
+const classNames = computed(() => [
+  fontClasses[props.size],
+  toVariants({
+    accent: props.accent,
+    variant: props.variant,
+    size: props.size,
+    busy: props.busy,
+    disabled: isDisabled.value,
+  }),
+])
+</script>
+
+<style lang="postcss" scoped>
+/* IMPLEMENTATION */
+.ui-button {
+  display: inline-flex;
+  align-items: center;
+  justify-content: center;
+  cursor: pointer;
+  vertical-align: middle;
+  gap: 0.75em;
+  border-width: 0.1rem;
+  outline: none;
+  border-style: solid;
+
+  &.busy,
+  &.disabled {
+    cursor: not-allowed;
+  }
+
+  /* ACCENT + VARIANT */
+
+  &.accent--brand {
+    &.variant--primary {
+      background-color: var(--color-normal-item-base);
+      border-color: var(--color-normal-item-base);
+      color: var(--color-normal-txt-item);
+
+      &:is(:hover, .hover, :focus-visible) {
```

Comment:
Given the last changes in the guidelines and how we handle variants, I think the `.hover` class is no longer needed and can be removed:
```suggestion
      &:is(:hover, :focus-visible) {
```
In addition, I think it would be great to add the `:focus-visible` styles defined in Figma, as we now have more information about this.